storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2638,6 +2647,34 @@ private int getNumOfAckerExecs(Map<String, Object> totalConf, StormTopology topo
         }
     }
 
+    private void upsertWorkerTokensInCreds(Map<String, String> creds, String user, String topologyId) {
+        if (workerTokenManager != null) {
```

Comment:
updateWorkerTokensInCreds ?

Reply:
from https://en.wikipedia.org/wiki/Merge_(SQL)
"A relational database management system uses SQL MERGE (also called upsert) statements to INSERT new records or UPDATE existing records depending on whether condition matches."

Reply:
Okey，got your idea